src/libtools/threads.c

```diff
@@ -251,8 +251,8 @@ static void* pthread_routine(void* p)
 #define PTHREAD_ATTR_UNALIGN(A)
 #define PTHREAD_ATTR(A) 	A
 #else
-#define PTHREAD_ATTR_ALIGN(A) pthread_attr_t aligned_attr = {0}; if(A) memcpy(&aligned_attr, A, 56)
-#define PTHREAD_ATTR_UNALIGN(A) if(A) memcpy(A, &aligned_attr, 56)
+#define PTHREAD_ATTR_ALIGN(A) pthread_attr_t aligned_attr = {0}; if(A) memcpy(&aligned_attr, A, sizeof(pthread_attr_t))
+#define PTHREAD_ATTR_UNALIGN(A) if(A) memcpy(A, &aligned_attr, sizeof(pthread_attr_t))
```

Comment:
This is unnecessary, but it has mo harm anyway.

Reply:
I'm not sure this is correct. Especially the "unalign" part: a pthread_attr_t is 56 bytes on x86_64, if you copy more than 56 bytes (like 64 on arm64), you will overwrite some memory not belonging to the pthread_attr_t

So, Im' pretty sure you will break things with this change.

Reply:
Yes, they are bigger so they can handled some miss-alignment.
But here, the it will be created on the stack and will be correctly aligned because of that, so the extrat space is not use and so the actual content is on front and can be just copy back and forth with no issue.